c/src/lib.rs

```diff
@@ -148,6 +148,21 @@ pub extern fn dataset_create(root_ptr: *const c_char, header_ptr: *mut Header) -
     }
 }
 
+#[no_mangle]
+pub extern fn file_compress(src_path_ptr: *const c_char, dst_path_ptr: *const c_char) {
```

Comment:
Need to check for null pointers. Return status code (i.e., non-zero value for error), so that the user does not have to manually query the last error to check if everything went well.